lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -243,6 +206,7 @@ getBlocks logic enqueue nodeId tipHeader checkpoints = do
         -- TODO don't be so wasteful.
```

Comment:
It makes sense, but it can't be done without modifying protocol. The performance difference is something like `150mcs` for a header in request (300ms for 2k blocks requested), which is significant. Please, create the issue for it and put its id here.

Reply:
We've already got @karknu working on that. Indeed it will be a protocol change. Hopefully the `InSpec`/`OutSpec` thing will make that painless.

Reply:
Alright, then we _must_ put the issue id here :)

Reply:
Ok I'll put it inline. It's this one https://iohk.myjetbrains.com/youtrack/issue/CSL-2148